File: Domain/Entities/Appointment.cs
namespace Domain.Entities
{
    public enum AppointmentStatus
    {
        Scheduled,
        Completed,
        Canceled
    }
    public class Appointment
    {
        public Guid AppointmentId { get; set; }
        public Guid PatientId { get; set; }
        public Guid ProviderId { get; set; }
        public DateTime AppointmentDate { get; set; }
        public string Reason { get; set; } = string.Empty;
        public AppointmentStatus Status { get; set; } = AppointmentStatus.Scheduled; //default value
  }
}

File: health-front/src/app/models/appointment.ts
export enum AppointmentStatus {
  Scheduled = 0,
  Completed = 1,
  Cancelled = 2
}

export interface Appointment {
  appointmentId?: string;
  patientId: string;
  providerId: string;
  patientName?: string;
  doctorUsername?: string;
  appointmentDate: Date;
  reason: string;
  status: AppointmentStatus;
}

export interface AppointmentResponse extends Appointment {
  id?: string;
}

export interface PaginatedResponse {
  items: Appointment[];
  totalCount: number;
  page: number;
  pageSize: number;
}